### PR TITLE
Initial scaffold for an operator

### PR DESCRIPTION
**Pull request by nerd0**

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+
+# Force dep to vendor the code generators, which aren't imported just used at dev time.
```
Comment (CK-Ward) [inline Gopkg.toml]: omg TG we are using `dep` 🎉

File: .gitignore [hunk 2]
```diff
@@ -10,3 +10,6 @@
 
 # Output of the go coverage tool, specifically when used with LiteIDE
 *.out
+
```
Comment (prateek) [inline .gitignore]: avoid including `vendor/` code in the repo, makes life a lot simpler in the long run.

Comment (nerd0) [reply .gitignore]: Fair, I was following the practices of the private internal repositories

**Comment by CK-Ward** [comment]
Are we trying to merge this anytime soon or is it a WIP?

File: .gitignore (hunk 2 above)
Comment (prateek) [reply .gitignore]: Yeah fair enough. We include vendor'd deps in the internal repos for build reproducibility, so fortunately we can avoid including them upstream (GH/etc)

File: tmp/build/build.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
```
Comment (prateek) [inline tmp/build/build.sh]: nit: weird to include a dir called `tmp/`, maybe separate the build scripts into a folder separate from the target output directory?

Comment (nerd0) [reply tmp/build/build.sh]: Agreed and removed.

**Comment by nerd0** [comment]
@CK-Ward This is very much a work in progress. I would like to get a single M3DB node launched with the operator before merging this in.

**Pull request closed without merging**